core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -184,6 +184,24 @@ class SocketServerTest extends JUnitSuite {
   }
 
   @Test
+  def testConnectionId() {
+    val socket1 = connect(protocol = SecurityProtocol.PLAINTEXT)
+    val socket2 = connect(protocol = SecurityProtocol.PLAINTEXT)
+    val serializedBytes = producerRequestBytes
+
+    sendRequest(socket1, serializedBytes)
+    sendRequest(socket2, serializedBytes)
+    val request1 = server.requestChannel.receiveRequest(2000)
+    val request2 = server.requestChannel.receiveRequest(2000)
+    val conn1Uniquifier = request1.connectionId.split("-")(2)
+    val conn2Uniquifier = request2.connectionId.split("-")(2)
+    assertNotEquals(conn1Uniquifier, conn2Uniquifier)
```

Comment:
Should we have assertions checking the exact expected value? Since there are only 2 requests, verifying that they are not the same seems a bit weak. We may also want to have 3 requests, just to be sure.